HW2/P2/parallel_vector.pyx

```diff
@@ -80,11 +80,18 @@ cpdef move_data_fine_grained(np.int32_t[:] counts,
    # data movement.  Be sure to avoid deadlock, and double-locking.
    ##########
    with nogil:
-       for r in range(repeat):
-           for idx in range(src.shape[0]):
-               if counts[src[idx]] > 0:
-                   counts[dest[idx]] += 1
-                   counts[src[idx]] -= 1
+     for r in range(repeat):
+         for idx in range(src.shape[0]):
```

Comment:
Yeah, I think the outer for loop should read something like: 
     for r in prange(repeat, num_threads=4):